docs/rules/group-exports.md

```diff
@@ -0,0 +1,62 @@
+# group-exports
+
+Reports when multiple named exports or CommonJS assignments are present in a single file and when the default export is not adjacent to the named export.
+
+**Rationale:** An `export` declaration or `module.exports` assignment can appear anywhere in the code. By requiring a single export declaration along with the default export being placed next to the named export, your exports will remain at one place, making it easier to see what exports a module provides.
+
+## Rule Details
+
+This rule warns whenever a single file contains multiple named exports or assignments to `module.exports` (or `exports`) and when the default export is not adjacent to the named export.
+
+### Valid
+
+```js
+// Default export statement is adjacent to named export -> ok
+export default function test() {}
+// A single named export statement -> ok
+export const valid = true
+```
+
+```js
+const first = true
+const second = true
+
+// A single named export statement -> ok
+export {
+  first,
+  second,
+}
+```
+
+```js
+// A single exports assignment -> ok
+module.exports = {
+  first: true,
+  second: true
+}
+```
+
+### Invalid
+
+```js
+// Multiple named export statements -> not ok!
+export const first = true
+export const second = true
+```
+
+```js
+// Default export is not adjacent to the named export statement -> not ok!
+export default {}
+const first = true
+export { first }
+```
+
+```js
+// Multiple exports assignments -> not ok!
+exports.first = true
+exports.second = true
```

Comment:
@ljharb assignments to `exports` and `module.exports` are indeed treated the same. 😄 I have added more tests to validate this.

So

```js
exports.first = true
module.exports.second = true
```

will cause this rule to issue a warning for both lines.

Currently the rule will ignore situations in which you assign something else than an object literal to `module.exports` (the rule will ignore all CommonJS module operations), so

```js
module.exports = function test() {}
module.exports.extra = true
module.exports.field = true
```

will not cause any warnings. Do you think it should warn even in situations like these?